src/block-data-providers/bitcoin-core/provider.ts

```diff
@@ -86,9 +91,17 @@ export class BitcoinCoreProvider
             return;
         }
 
+        const networkInfo = await this.getNetworkInfo();
+        const verbosityLevel = this.hasFullVerboseInput(networkInfo.version)
+            ? 3
+            : 2;
```

Comment:
Let's create a class variable called `verbosityLevel` and set it inside `onApplicationBootstrap` instead of `sync`
The version of node software won't change in-between calls so fetching it over and over is useless IMHO.

Reply:
I understand, that is why i put it in the sync function, because there surely cant be a change of bitcoind version while in sync but it is logically possible to be switched after sync. Logically possible because all the config depends on is the RPC details.

Reply:
What do you mean by in-sync and after sync?

Reply:
I changed my mind, just leave it in there for now